.github/workflows/self-nightly-past-ci-caller.yml

```diff
@@ -0,0 +1,140 @@
+name: Self-hosted runner (nightly-past-ci-caller)
+
+on:
+  push:
+    branches:
+      - run_nightly_ci*
+      - run_past_ci*
+
+jobs:
+  build_nightly_ci_images:
+    name: Build Nightly CI Docker Images
+    if: (github.event_name == 'schedule') || ((github.event_name == 'push') && startsWith(github.ref_name, 'run_nightly_ci'))

```

Comment:
All this `if` condition is just to make the workflow run could be triggered in desired situations:

- schedule event: run nightly and past ci
- push event with `run_nightly_ci*` branch: run nightly ci
- push event with `run_past_ci*` branch: run past ci